manager/kitchen_manager.py

```diff
@@ -0,0 +1,202 @@
+    :param kitchens - an array of abstract class`s children objects
+    """
+    kitchens = []
+
+    @write_attribs_in_file
+    def add_kitchen(self, kitchen):
+        """
+        Method which adds kitchens to array
+        :param kitchen: name of objects in general
+        :return: an addition of kitchen to array
+        """
+        self.kitchens.append(kitchen)
+
+    @write_attribs_in_file
+    def find_bigger_than(self, size):
+        """
+        Method finds all objects bigger than given size
+        :param size: of place
+        :return: a filtered list with needed objects
+        """
+        return list(filter(lambda kitchen: kitchen.size > size, self.kitchens))
+
+    @write_attribs_in_file
+    def find_bigger_capacity_than(self, capacity):
+        """
+        Method finds all objects with bigger capacity than given size
+        :param capacity: of possible people in place
+        :return: a filtered list with needed objects
+        """
+        return list(filter(lambda kitchen: kitchen.capacity > capacity, self.kitchens))
+
+    def __len__(self):
+        """
+        Returns the number of kitchens in the manager
+        """
+        return len(self.kitchens)
+
+    def __getitem__(self, index):
+        """
+        Returns the kitchen at the specified index
+        """
+        return self.kitchens[index]
+
+    def __iter__(self):
+        """
+        Returns an iterator over the kitchens in the manager
+        """
+        return iter(self.kitchens)
+
+    @limit_of_cals
+    def type_of_kitchens(self):
+        """
+        :return: Type of kitchen for every kitchen
+        """
+        return [kitchen.kitchen_type() for kitchen in self.kitchens]  # TODO check kitchen_type(), why random?
+
+    @limit_of_cals
+    def enumerate_concatenation(self):
+        """
+        Returns a concatenation of each kitchen object and its index in the manager
+        """
+        return list(enumerate(self.kitchens))
+
+    @limit_of_cals
+    def zip_concatenation(self):
+        """
+        Returns a concatenation of each kitchen object and the result of the kitchen_type() method
+        """
+        return list(zip(self.kitchens, self.type_of_kitchens()))
+
+    @limit_of_cals
+    @write_attribs_in_file
+    def rating_check(self, minimal_rating):
+        """
+        Checks if all objects in the manager satisfy the given condition
+        """
+        result = [kitchen.rating > minimal_rating for kitchen in self.kitchens]
+        return {
+            "all": all(result),
+            "any": any(result)}
+
+
+class SetManager:
+    """
+    Class to work with chefs inside every kitchen
+    """
+    all_chefs = []
+
+    def __init__(self, regular_manager):
+        """
+        Put chefs from all kitchens in one list
+        :param regular_manager: object type Manager
+        """
+        self.all_chefs = [chef for kitchen in regular_manager.kitchens for chef in kitchen]
```

Comment:
the problem is that there would be duplicates. For example, if you have two kitchens with same type, their chefs would be same